daemon/src/position_metrics.rs

```diff
@@ -0,0 +1,344 @@
+                ..self
+            },
+            CetTimelockExpiredPriorOracleAttestation
+            | CetTimelockExpiredPostOracleAttestation { .. } => Self {
+                is_open: false,
+                is_closed: true,
+                ..self
+            },
+            RefundTimelockExpired { .. } => Self {
+                // a rollover with an expired timelock should be rejected for settlement and
+                // rollover, hence, this is closed
+                is_open: false,
+                is_closed: true,
+                ..self
+            },
+            OracleAttestedPriorCetTimelock { .. } | OracleAttestedPostCetTimelock { .. } => Self {
+                // we know the closing price already and can assume that the cfd will be closed
+                // accordingly
+                is_open: false,
+                is_closed: true,
+                ..self
+            },
+        }
+    }
+}
+
+impl db::ClosedCfdAggregate for Cfd {
+    fn new_closed(_: Self::CtorArgs, closed_cfd: db::ClosedCfd) -> Self {
+        let db::ClosedCfd {
+            id,
+            position,
+            n_contracts,
+            ..
+        } = closed_cfd;
+
+        let quantity_usd =
+            Usd::new(Decimal::from_u64(u64::from(n_contracts)).expect("u64 to fit into Decimal"));
+
+        Self {
+            id,
+            position,
+            quantity_usd,
+
+            is_open: false,
+            is_closed: true,
+            is_failed: false,
+            is_refunded: false,
+            version: 0,
+        }
+    }
+}
+
+mod metrics {
+    use crate::position_metrics::Cfd;
+    use model::Position;
+    use model::Usd;
+    use rust_decimal::prelude::ToPrimitive;
+    use std::collections::HashMap;
+    use std::iter::Filter;
+    use std::slice::Iter;
+
+    const POSITION_LABEL: &str = "position";
+    const POSITION_LONG_LABEL: &str = "long";
+    const POSITION_SHORT_LABEL: &str = "short";
+
+    const STATUS_LABEL: &str = "status";
+    const STATUS_OPEN_LABEL: &str = "open";
+    const STATUS_CLOSED_LABEL: &str = "closed";
+    const STATUS_FAILED_LABEL: &str = "failed";
+    const STATUS_REFUNDED_LABEL: &str = "refunded";
+    // this is needed so that we do not lose cfds which are in a weird state, e.g. open & closed.
+    // This should be 0 though but for the time being we add it
+    const STATUS_UNKNOWN_LABEL: &str = "unknown";
+
+    static POSITION_QUANTITY_GAUGE: conquer_once::Lazy<prometheus::GaugeVec> =
+        conquer_once::Lazy::new(|| {
+            prometheus::register_gauge_vec!(
+                "positions_quantity",
+                "Total quantity of positions on ItchySats.",
+                &[POSITION_LABEL, STATUS_LABEL]
+            )
+            .unwrap()
+        });
+
+    static POSITION_AMOUNT_GAUGE: conquer_once::Lazy<prometheus::IntGaugeVec> =
+        conquer_once::Lazy::new(|| {
+            prometheus::register_int_gauge_vec!(
+                "positions_total",
```

Comment:
```suggestion
                "positions_number",
```

`positions_total_number` would work too if you prefer that.

"total" is a bit ambiguous, I actually thought it referred to the quantities.

Reply:
mhmm.. not sure about that. 
I tried to follow the naming convention here: https://prometheus.io/docs/practices/naming/

Reply:
how about `positions_number_total` ? that's less ambiguous, and shouldn't really get into the way of the naming practices?

Reply:
Done